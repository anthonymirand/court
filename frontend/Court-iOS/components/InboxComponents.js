import React from 'react';
import PropTypes from 'prop-types';
import { Platform, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import Avatar from '../components/Avatar';

/**
* Card in the user's inbox for each of their current chats
*/
export class InboxItem extends React.Component {
  onPressChat = () => {
    const { animalName, color, imgUrl, name, lastMessage, lastTime, percent} = this.props;
    // This is where we should handle navigation to the chat screen
    this.props.onPress(this.props.name, {name: name, animalName: animalName, color: color, imgUrl: imgUrl});
  }

  render() {
    const { animalName, color, imgUrl, name, lastMessage, lastTime, percent} = this.props;
    return (
      <TouchableOpacity activeOpacity={0.5} onPress={this.onPressChat}>
        <View style={styles.InboxCard}>
          <View style={styles.avatarWrapper}>
            <Avatar width={65} imgURL={imgUrl} animalName={animalName} color={color}/>
          </View>
          <View style={styles.inboxTextWrapper}>
            <Text style={styles.nameStyle} numberOfLines={1}>{name}</Text>
            <Text style={styles.messageStyle} numberOfLines={1}>{lastMessage}</Text>
            <Text style={styles.timeStyle}>{lastTime}</Text>
          </View>
          <View >
            <Text style={styles.percentStyle}>{percent}%</Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

InboxItem.propTypes = {
  /**
  * URL for the given user's profile picture
  */
  imgUrl: PropTypes.string,
  /**
  * Name of the other user in the chat
  */
  name: PropTypes.string.isRequired,
  /**
  * The text of the last message sent in the chat
  */
  lastMessage: PropTypes.string.isRequired,
  /**
  * The time that the last message was sent
  */
  lastTime: PropTypes.string.isRequired,
  /**
  * Percent of profile unlocked for the given user
  */
  percent: PropTypes.string.isRequired,
}

const styles = StyleSheet.create({
  InboxCard: {
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { height: 8 },
        shadowOpacity: 0.15,
        shadowRadius: 9,
      },
      android: {
        elevation: 20,
      },
    }),
    backgroundColor: 'white',
    paddingHorizontal: 10,
    paddingVertical: 15,
    marginVertical: 12,
    marginHorizontal: 20,
    borderRadius: 15,
    flex: 1,
    flexDirection: 'row',
  },
  avatarWrapper:{
    flexDirection: 'column',
    justifyContent: 'center',
  },
  inboxTextWrapper: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    paddingHorizontal: 10,
  },
  nameStyle: {
    fontSize: 25,
    fontFamily: 'orkney-medium',
    color: '#21ACA5',
  },
  messageStyle: {
    fontFamily: 'orkney-regular',
    fontSize: 15,
  },
  timeStyle: {
    paddingTop: 8,
    fontFamily: 'orkney-light',
    color: 'grey',
  },
  percentStyle: {
    fontSize: 25,
    fontFamily: 'orkney-medium',
    color: '#21ACA5',
  },
  contentContainer: {
    paddingTop: 30,
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    backgroundColor: 'green',
    flexDirection: 'column',
  }
});
