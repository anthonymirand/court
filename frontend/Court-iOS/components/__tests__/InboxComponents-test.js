import 'react-native';
import React from 'react';
import { InboxItem } from '../InboxComponents';
import renderer from 'react-test-renderer';

it('Renders Default InboxCard', () => {
	const lastMessage = {text: 'I like the office', createdAt: '12:40 PM GMT'};
	const profile = {animal: 'bear', color: 'teal', first_name: 'John', percent_unlocked: 12};
  const tree = renderer.create(
    <InboxItem
      profile={profile}
      lastMessage={lastMessage}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
