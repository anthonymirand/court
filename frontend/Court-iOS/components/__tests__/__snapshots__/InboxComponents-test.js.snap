// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default InboxCard 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "row",
        "marginHorizontal": 20,
        "marginVertical": 12,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 8,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 9,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": 65,
            "width": 65,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 0,
                "overflow": "hidden",
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://test.com/img.png",
              }
            }
            style={
              Object {
                "height": 65,
                "width": 65,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.03)",
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#21ACA5",
            "fontSize": 25,
          }
        }
      >
        River
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        Test Message
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "grey",
            "paddingTop": 8,
          }
        }
      >
        10:00 AM
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#21ACA5",
            "fontSize": 25,
          }
        }
      >
        60%
        %
      </Text>
    </View>
  </View>
</View>
`;
